BackendTaskPilot/Program.cs

```diff
@@ -0,0 +1,38 @@
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+var builder = WebApplication.CreateBuilder(args);
+var app = builder.Build();
+
+ExchangeRateService service = new ExchangeRateService();
+
+// REST API to get exchange rates based on input currencies
+app.MapPost("/", service.GetExchangeRates);
+
+app.Run();
+
+public struct CnbRatesResponse
+{
+    [JsonPropertyName("rates")] public List<Rate> Rates { get; set; }
+}
+
+public struct Rate
+{
+    [JsonPropertyName("currencyCode")] public string CurrencyCode { get; set; }
+    [JsonPropertyName("rate")] public float RateValue { get; set; }
+}
+
+internal class ExchangeRateService
+{
+    private const string MainCurrenciesUrl = "https://api.cnb.cz/cnbapi/exrates/daily";
+
+    public IEnumerable<Rate> GetExchangeRates(List<string> currencies)
+    {
+        HttpClient client = new HttpClient();
+			
+        var data = client.GetStringAsync(MainCurrenciesUrl).Result;
+        var response = JsonSerializer.Deserialize<CnbRatesResponse>(data);
+
+        return response.Rates.Select(rate => rate).Where(rate => currencies.Contains(rate.CurrencyCode));
+    }
```

Comment:
Think about some error handling. Add some logs for when the data is null as you don't want to perform the rest of the work if there are no rates returned as it could lead to NullRefrenceExceptions. Also when we request rates for multiple currencies we can add some logs of Warning type for the currencies that haven't been returned. I have noticed that rates for some required currencies aren't being returned, perhaps will need to think of an alternative endpoint